src/daft-logical-plan/src/optimization/optimizer.rs

```diff
@@ -90,7 +103,7 @@ pub struct Optimizer {
 
 impl Optimizer {
     pub fn new(config: OptimizerConfig) -> Self {
```

Comment:
I think it may be worthwhile to refactor this to a `OptimizerBuilder` where we can call methods to add batches.

```
let builder = OptimizerBuilder::default()
  .join_reordering()
  .simplify_expressions();

let optimizer = builder.build()
```

Reply:
Sounds good, done!